src/util.rs

```diff
@@ -81,7 +80,10 @@ pub fn field_set(cx: &ExtCtxt, parent: P<ast::Expr>, key: &[Spanned<ast::Ident>]
     assert!(key.len() > 0);
 
     let parent = if key.len() > 1 {
-        field_get(cx, parent, key.init(), true)
+        match key.split_last() {
+            None => panic!("Contradiction to assertion"),
+            Some(split_array) => field_get(cx, parent, split_array.1, true),
```

Comment:
Do something in the lines of (not tested)
```rust
let ident, parent = match key.split_last() {
    None => panic!("At least one ident is required"),
    Some((ident, []) => (ident, parent),
    Some((ident, path)) => field_get(cx, parent, ident, true),
}
let accessor = field_accessor(cx, ident, Accessor::Set);
```
And remove the now redundant assert